generators/wizard/templates/wizard.ts

```diff
@@ -99,27 +118,68 @@ export class <%= wizard.class %> extends<% if (wizard.isExtendingMes) { %> MesCo
 
     //#region Public methods
 
-    public ngOnChanges(changes: ng.SimpleChanges): void {
-
-    }
-
+    /**
+     * Method that prepares the data for the wizard
+     */
     public prepareDataInput(): Promise<Cmf.Foundation.BusinessOrchestration.BaseInput[]> {
         // Please provide all the inputs objects that need to be fetched when the wizard loads (if required)
         const inputs: Cmf.Foundation.BusinessOrchestration.BaseInput[] = [];
+        const instanceInput: Cmf.Foundation.BusinessOrchestration.GenericServiceManagement.InputObjects.GetObjectByIdInput
+            = new Cmf.Foundation.BusinessOrchestration.GenericServiceManagement.InputObjects.GetObjectByIdInput();
+        instanceInput.IgnoreLastServiceId = true;
+        instanceInput.Id = this._pageBag.context.instance.Id;
+        // TODO: If you know your EntityType just assign its name to the type below. Example: "Material"
+        instanceInput.Type = this.framework.caches.entityTypes.getEntityTypeNameFromInstance(this.instance);
         return Promise.resolve(inputs);
     }
 
+    /**
+     * Method that receive the data from prepareDataInput
+     */
     public handleDataOutput(outputs: Cmf.Foundation.BusinessOrchestration.BaseOutput[]): Promise<void> {
         // Please handle all the outputs from prepareDataInput (if required)
+        if (outputs != null && outputs.length > 0) {
+            const output: Cmf.Foundation.BusinessOrchestration.GenericServiceManagement.OutputObjects.GetObjectByIdOutput =
+                <Cmf.Foundation.BusinessOrchestration.GenericServiceManagement.OutputObjects.GetObjectByIdOutput>outputs[0];
+
+            if (output != null && output.Instance != null) {
+                this.instance = output.Instance;
+                this.instanceForDisplay = output.Instance;
+                if (this._nestedWizard != null) {
+                    return this._nestedWizard.reEvaluateContextPreConditions({ 'instance': this.instance }).
+                        then(() => {
+                            return Promise.resolve();
+                        });
+                }
+            } else {
+                this.instance = null;
+            }
+        } else {
+            this.instance = null;
+        }
+
+        if (this.instance == null) {
+            throw new Error("No Instance Found"); // Missing i18n translation
```

Comment:
You can add the error string in i18n and remove the TODO here
```suggestion
            throw new Error(i18n.errors.NO_INSTANCE);
```